rclcpp/src/rclcpp/node_interfaces/node_logger.cpp

```diff
@@ -0,0 +1,34 @@
+// Copyright 2017 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "rclcpp/node_interfaces/node_logger.hpp"
+
+using rclcpp::node_interfaces::NodeLogger;
+
+NodeLogger::NodeLogger(rclcpp::node_interfaces::NodeBaseInterface * node_base)
+: node_base_(node_base)
+{
+  // TODO(dhood): use the namespace (slashes converted to dots)
```

Comment:
This would be nice to have in too, since that's part of the point of having the node logger right? So users don't have to do something like `convert_slash_to_dot(node->get_namespace()) + node->get_name()` for every logging call? (but still get the benefit of better namespaces in logging by default)

I understand this is higher risk to merge though...

Reply:
It's definitely planned, I agree that it is the main point of the node logger. My thinking was to put the "logger name from node name/namespace" in rcl and then wrap it with python as well. Because of the reach of that change and the fact that things at the user level shouldn't change once we make the swap under the hood though, I consider it a separate "step" and so scoped it out of this PR for time reasons. next step though :smile:

Reply:
It's definitely planned, I agree that it is the main point of the node logger. My thinking was to put the "logger name from node name/namespace" in rcl and then wrap it with python as well. Because of the reach of that change and the fact that things at the user level shouldn't change once we make the swap under the hood though, I consider it a separate "step" and so scoped it out of this PR for time reasons. It'll be the very next step though :smile:

Reply:
Following up: https://github.com/ros2/rclcpp/pull/433